### DISCUSSION TITLE
Authentication Methods Unclear

### DISCUSSION THREAD

**Discussion by dwhagar** [Kozea/Radicale#1459] [Ideas]
So, is the _only_ authentication method that is actually usable, `htpasswd`?

The config file that comes with the distribution clearly states:

    [auth]
    # Authentication method
    # Value: None | htpasswd | IMAP | LDAP | PAM | courier | http

Of course, setting the `type=` to IMAP, LDAP, or PAM shows that there are no Python files for those authentication methods.  Taking a look at the actual code, it clearly shows that such a module has yet to be written.  From what I read on another issue, this has been intentionally done but the config file suggests it has not?  What gives?  Why was this removed?  Is there a workaround?

**Comment by Tntdruid** [comment]
Can only use htpasswd, the OS dist is a outdated version.

http://radicale.org/setup/

**Comment by kousu** [comment]
Hiya,

I'm also wondering what the plan here is. Would you be open to accepting a new Radicale 3.0-compatible `auth/pam.py` and `auth/imap.py` files? As recently as half a year before this thread (https://github.com/Kozea/Radicale/issues/552#issuecomment-281982826) PAM was still supported.

**Comment by pbiering** [comment]
@kousu : if required, please submit PR for authentication using PAM or IMAP against latest 3.x

**Comment by kousu** [comment]
So, personally, I solved this by writing my own auth module: https://github.com/kousu/radicale-bsdauth
But it's pretty niche. There's probably not too many people using OpenBSD and radicale together. But if you want to try it, go for it! I've been using it for two years now without a peep of trouble.

I would like to see PAM support again though, or especially LDAP. That would be really cool. It was pretty straightforward to write that module as an extra plugin, so if anyone else wants to work a non-core LDAP module you can @ me for help.

**Comment by pbiering** [comment]
Sophisticated user authentication should be handover'ed to reverse proxy, as they have a bunch of authentication options already build-in or available via plug-in modules.
